src/AndersonAcceleration.f90

```diff
@@ -85,109 +85,56 @@ MODULE AndersonAccelerationTypes
   ENDIF
 
-  n=0
-  nlocal=0
-  depth=-1
-  beta=0.0_SRK
-  start=1
   !Pull Data from Parameter List
-  CALL validParams%get('AndersonAccelerationType->n',n)
-  CALL validParams%get('AndersonAccelerationType->nlocal',nlocal)
-  CALL validParams%get('AndersonAccelerationType->depth',depth)
-  CALL validParams%get('AndersonAccelerationType->beta',beta)
-  CALL validParams%get('AndersonAccelerationType->start',start)
-  CALL validParams%clear()
+  CALL Params%get('AndersonAccelerationType->N',solver%N)
+  CALL Params%get('AndersonAccelerationType->depth',solver%depth)
+  CALL Params%get('AndersonAccelerationType->beta',solver%beta)
+  CALL Params%get('AndersonAccelerationType->start',solver%start)
 
   IF(.NOT. solver%isInit) THEN
-    IF(n < 1) THEN
-      CALL eAndersonAccelerationType%raiseError('Incorrect input to '// &
-          modName//'::'//myName//' - Number of values (n) must be '// &
-          'greater than 0!')
-    ELSE
-      solver%n=n
-    ENDIF
-
-    IF((nlocal < 1) .AND. (nlocal > n)) THEN
-      CALL eAndersonAccelerationType%raiseError('Incorrect input to '// &
-          modName//'::'//myName//' - Number of values (nlocal) must be '// &
-          'greater than 0 and less than or equal to (n)!')
+    IF(solver%N < 1) THEN
+      CALL eAndersonAccelerationType%raiseError('Incorrect input to '//modName// &
+          '::'//myName//' - Number of unkowns (N) must be greater than 0!')
     ENDIF
 
-    IF(depth < 0) THEN
-      CALL eAndersonAccelerationType%raiseError('Incorrect input to '// &
-          modName//'::'//myName//' - Depth values (depth) must be '// &
-          'greater than or equal to 0!')
-    ELSE
-      solver%depth=depth
-    ENDIF
-
-    IF(start <= 0) THEN
-      CALL eAndersonAccelerationType%raiseError('Incorrect input to '// &
-          modName//'::'//myName//' - Starting iteration must be '// &
-          'greater than to 0!')
-    ENDIF
-
-    IF((beta<=0.0_SRK) .OR. (beta>1.0_SRK)) THEN
-      CALL eAndersonAccelerationType%raiseError('Incorrect input to '// &
-          modName//'::'//myName//' - Beta must be '// &
-          'greater than 0 and less than or equal to 1!')
-    ELSE
-      solver%beta=beta
-    ENDIF
-
-#ifdef FUTILITY_HAVE_Trilinos_NOX
-    CALL tmpPL%clear()
-    CALL tmpPL%add('VectorType->n',n)
-    CALL tmpPL%add('VectorType->MPI_Comm_ID',solver%MPIparallelEnv%comm)
-    CALL tmpPL%add('VectorType->nlocal',nlocal)
-    CALL tmpPL%add('VectorType->engine',VM_TRILINOS)
-    CALL VectorFactory(solver%X,tmpPL)
-    CALL solver%X%set(0.0_SRK)
-
-    SELECTTYPE(x=>solver%X); TYPE IS(TrilinosVectorType)
-      CALL Anderson_Init(solver%id,solver%depth,solver%beta,start,x%b)
-    ENDSELECT
-#else
-    CALL eAndersonAccelerationType%raiseError(modName//'::'//myName// &
-        ' - NOX (Trilinos) is not present in build')
-#endif
+    !Allocate member arrays
+    M=solver%depth+1
+    ALLOCATE(solver%x(solver%N,M),solver%Gx(solver%N,M),solver%r(solver%N,M), &
```

Comment:
do we really need to store x, Gx, and r?  Is that saving flops or is that just making life easier?

Reply:
In the case of r, saves a lot of flops...well it's a simple problem compared to the actual simulation costs but cuts the number of flops by a significant fraction. Also you have no choice with regard to Gx and x. You have to save the last depth+1 vectors.